### PR TITLE
Suppress warning if UnconditionalSuppressMessageAttribute is used

### PR DESCRIPTION
**Pull request by agocke**
If/when Roslyn ships a universal suppress behavior here we can remove this code.

### CODE REVIEW

File: src/ILLink.RoslynAnalyzer/AnalyzerUtils.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+			}
+
+			return true;
+		}
+
+		const string SuppressMessageFqn = "System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute";
+
+		internal static bool IsDiagnosticSuppressed(OperationAnalysisContext operationContext, string diagnosticId)
+		{
+			var attributes = operationContext.ContainingSymbol.GetAttributes();
```
Comment (mateoatr) [inline src/ILLink.RoslynAnalyzer/AnalyzerUtils.cs:39]: Will this work when the suppression is up in the tree?
e.g.
```C#
[UnconditionalSuppressMessage ("", "IL1234:MyWarning")]
class Foo {
   void Bar () {
     // Triggers warning IL1234
  }
}
```

**Review by mateoatr — COMMENTED** [review]
What about unconditional suppressions that use the target property?

File: src/ILLink.RoslynAnalyzer/AnalyzerUtils.cs (hunk 1 above)
Comment (agocke) [reply src/ILLink.RoslynAnalyzer/AnalyzerUtils.cs:39]: Right, this only handles local suppression. Global suppression support will have to be added in another PR.

**Comment by agocke** [comment]
As far as I can tell, the Target property is only relevant on attributes higher than the member level.

**Comment by agocke** [comment]
I'm gonna close this for now, since it looks like the Roslyn change is gonna go through.

**Comment by tlakollo** [comment]
Closing in favor of https://github.com/dotnet/roslyn/pull/51792

**Pull request closed without merging**